elytron/src/main/java/org/wildfly/extension/elytron/AuthenticationContextAssociationProcessor.java

```diff
@@ -17,14 +17,24 @@
  */
 package org.wildfly.extension.elytron;
 
+import static java.security.AccessController.doPrivileged;
```

Comment:
This import should really be org.wildfly.extension.elytron.SecurityActions.doPrivileged that way a WildFlySecurityManager.isChecking() call will be made to check if we really do need the AccessController.doPrivileged().

Reply:
This is only every run once, so it does not make any real difference.

Reply:
Once per first request to each deployment whilst parallel requests are waiting for the lock.